tasks/src/lib.rs

```diff
@@ -342,6 +380,52 @@ pub enum SchedulerError {
     /// It's better to halt the execution to avoid damage.
     #[error("unexpected tag in command completion result")]
     UnexpectedTaggedResponse(Tagged<'static>),
+
+    #[error("unexpected BYE response")]
+    UnexpectedByeResponse(Bye<'static>),
+
+    #[error("unexpected BAD response: {}", .0.text)]
+    UnexpectedBadResponse(StatusBody<'static>),
+
+    #[error("unexpected NO response: {}", .0.text)]
+    UnexpectedNoResponse(StatusBody<'static>),
+
+    #[error("missing required data for command {0}")]
+    MissingData(String),
```

Comment:
I think we need to be a bit careful here to avoid already getting in the weeds of IMAP command semantics.

`UnexpectedTaggedResponse` is a `SchedulerError` because it signals that `imap-tasks` lost track of connection understanding. This could be a severe error during scheduling.

The variants we want to add here are rather task specific, though. Maybe we should create a dedicated enum that holds these common errors? This would also allow us to do refinements (as explained above).

I'm not sure yet about `UnexpectedByeResponse`...